src/Bundle/test/src/Grid/BookByAmericanAuthorsGrid.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace App\Grid;
+
+use App\Entity\Book;
+use App\Entity\Nationality;
+use Sylius\Bundle\GridBundle\Builder\Field\StringField;
+use Sylius\Bundle\GridBundle\Builder\Filter\Filter;
+use Sylius\Bundle\GridBundle\Builder\GridBuilderInterface;
+use Sylius\Bundle\GridBundle\Grid\AbstractGrid;
+use Sylius\Bundle\GridBundle\Grid\ResourceAwareGridInterface;
+
+class BookByAmericanAuthorsGrid extends AbstractGrid implements ResourceAwareGridInterface
+{
+    public static function getName(): string
+    {
+        return 'app_book_by_american_authors';
+    }
+
+    public static function getResourceClass(): string
+    {
+        return Book::class;
+    }
+
+    public function buildGrid(GridBuilderInterface $gridBuilder): void
+    {
+        $gridBuilder
+            ->setRepositoryMethod('createAmericanBooksQueryBuilder')
+            ->addFilter(Filter::create('title', 'string'))
+            ->addFilter(Filter::create('author', 'entity')
+                ->setFormOptions([
+                    'class' => Nationality::class,
+                ])
+            )
+            ->addFilter(Filter::create('nationality', 'entity')
+                ->setOptions([
+                    'fields' => ['author.nationality'],
+                ])
+                ->setFormOptions([
+                    'class' => Nationality::class,
```

Comment:
We should extend this functionality to provide possibility to put parameter here